imports/plugins/included/shipping-rates/client/templates/settings/rates.js

```diff
@@ -0,0 +1,228 @@
+          result.push(method.methods);
+        }
+      }
+      return result[0];
+    }
+
+    // return shipping Grid
+    return {
+      component: MeteorGriddle,
+      publication: "Shipping",
+      transform: transform,
+      collection: Shipping,
+      matchingResultsCount: "shipping-count",
+      showFilter: true,
+      useGriddleStyles: false,
+      rowMetadata: customRowMetaData,
+      filteredFields: filteredFields,
+      columns: filteredFields,
+      noDataMessage: noDataMessage,
+      onRowClick: editRow,
+      columnMetadata: customColumnMetadata,
+      externalLoadingComponent: Loading
+    };
+  },
+
+  instance() {
+    const instance = Template.instance();
+    return instance;
+  },
+
+  shippingRate() {
+    const instance = Template.instance();
+    const id = instance.state.get("editingId");
+    const providerRates = Shipping.findOne({ "provider.name": "reaction-shipping-rates" }) || {};
+    let rate = {};
+    if (providerRates && providerRates.methods) {
+      if (id) {
+        for (method of providerRates.methods) {
+          if (method._id === id) {
+            rate = method;
+          }
+        }
+      } else {
+        // a little trick to provide _id for insert
+        rate._id = providerRates._id;
+      }
+    }
+    return rate;
+  }
+});
+
+//
+// on submit lets clear the form state
+//
+Template.shippingRatesSettings.events({
+  "submit #shipping-rates-update-form": function () {
+    const instance = Template.instance();
+    instance.state.set({
+      isEditing: false,
+      editingId: null
+    });
+  },
+  "submit #shipping-rates-insert-form": function () {
+    const instance = Template.instance();
+    instance.state.set({
+      isEditing: true,
+      editingId: null
+    });
+  },
+  "click .cancel, .shipping-grid-row .active": function () {
+    instance = Template.instance();
+    // remove active rows from grid
+    instance.state.set({
+      isEditing: false,
+      editingId: null
+    });
+    // ugly hack
+    $(".shipping-grid-row").removeClass("active");
+  },
+  "click .delete": function () {
+    const confirmTitle = i18next.t("admin.shippingSettings.confirmRateDelete");
+    const confirmButtonText = i18next.t("app.delete");
+    const instance = Template.instance();
+    const id = instance.state.get("editingId");
+    // confirm delete
+    Alerts.alert({
+      title: confirmTitle,
+      type: "warning",
+      showCancelButton: true,
+      confirmButtonText: confirmButtonText
+    }, (isConfirm) => {
+      if (isConfirm) {
+        if (id) {
+          Meteor.call("shipping/rates/delete", id);
+          instance.state.set({
+            isEditing: false,
+            editingId: null
+          });
+        }
+      }
+    });
+  },
+  "click .shipping-grid-row": function (event) {
+    // toggle all rows off, then add our active row
+    $(".shipping-grid-row").removeClass("active");
```

Comment:
Good point, I'll capture these two suggestions @prosf's hook in a quick update.. there's a few more thing cleanup items.